lib/task.ml

```diff
@@ -22,7 +22,15 @@ let do_task f p =
     | TasksActive -> raise e
     | _ -> ()
 
-let setup_pool ~num_additional_domains =
+let named_pools = Hashtbl.create 8
+
+let named_pool_mutex = Mutex.create ()
+
+let setup_pool ?name ~num_additional_domains () =
```

Comment:
Excuse my ignorance, but why do we need to add the unit `()` arg to the end of this one?

Reply:
Consider the case when we don't include the unit value argument:

```ocaml
# let setup_pool ?name ~num_additional_domains = ();;
Warning 16 [unerasable-optional-argument]: this optional argument cannot be erased.
val setup_pool : ?name:'a -> num_additional_domains:'b -> unit = <fun>
# setup_pool ~num_additional_domains:5;;
- : ?name:'a -> unit = <fun>
```
vs when we have it: 

```ocaml
# let setup_pool ?name ~num_additional_domains () = ();;
val setup_pool : ?name:'a -> num_additional_domains:'b -> unit -> unit =
  <fun>
# setup_pool ~num_additional_domains:5 ();;
- : unit = ()
```

https://stackoverflow.com/a/1667891 has more information on this issue.